Xamarin.Essentials/TextToSpeech/TextToSpeech.android.cs

```diff
@@ -0,0 +1,290 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.OS;
+using Android.Speech.Tts;
+using Xamarin.Essentials;
+
+namespace Xamarin.Essentials
+{
+    public static partial class TextToSpeech
+    {
+        const int maxSpeechInputLengthDefault = 4000;
+
+        static global::Android.Speech.Tts.TextToSpeech textToSpeech;
+
+        static TextToSpeech()
+        {
+            Task.Run(() => Initialize());
+
+            return;
+        }
+
+        static Java.Util.Locale localedefault = null;
+
+        static async Task Initialize()
+        {
+            // set up the TextToSpeech object
+            // third parameter is the speech engine to use
+            textToSpeech = new global::Android.Speech.Tts.TextToSpeech(Platform.CurrentContext, default(TextToSpeechListener), "com.google.android.tts");
+
+            var locales = await GetLocalesAsync();
+
+            return;
+        }
+
+        public static async Task SpeakAsync(string text, CancellationToken cancelToken = default(CancellationToken)) =>
+             await SpeakAsync(text, default(SpeakSettings), cancelToken);
+
+        public static async Task SpeakAsync(string text, SpeakSettings settings, CancellationToken cancelToken = default(CancellationToken))
+        {
+            if (string.IsNullOrEmpty(text))
+            {
+                throw new ArgumentNullException(nameof(text), "Text cannot be null or empty string");
+            }
+
+            Task initialization = null;
+            if (textToSpeech == null)
+            {
+                initialization = Initialize();
+            }
+
+            var parts = SplitText(text, maxSpeechInputLengthDefault);
+
+            await Task.Run(() =>
+            {
+                initialization?.Wait();
+
+                // set the speed and pitch
+                // textToSpeech.SetPitch(settings.Pitch.Value);
+                // textToSpeech.SetSpeechRate(settings.SpeakRate.Value);
+
+                var bundle = Bundle.Empty;
+                string stringutteranceid = null;
+
+                foreach (var t in parts)
+                {
+                    textToSpeech.Speak(t, QueueMode.Flush, bundle, stringutteranceid);
+                }
+            });
+
+            return;
+        }
+
+        private static List<string> SplitText(string text, int maxSpeechInputLengthDefault)
```

Comment:
I don't think we should do this. See https://github.com/xamarin/Essentials/issues/12#issuecomment-380824337